src/videojs-media-sources.js

```diff
@@ -301,9 +365,9 @@
           offset += segment.byteLength;
         });
 
-        // Set timestampOffset if we have been given one
-        if (this.timestampOffset !== undefined) {
-          destinationBuffer.timestampOffset = this.timestampOffset;
+        // Set timestampOffset if it was changed before this append
```

Comment:
I think that introduces the potential for more confusion and complicates reasoning about the code.

The way it is right now, `timestampOffset` is set in exactly *one* place and that place is right before we `appendBuffer` which is exactly how a user would use `timestampOffset` in normal MSE code (ie. the `timestampOffset` is pretty much set exactly where one would expect it to be set.)